File: Hw7/code/161044096_değer_mandal.pl
% Prolog is a declarative programming language, because declarative
% programming refers to programming with recipe so you tell machine what
% you want, machine make the environment itself.


mother(X,Y):-
	female(X),
	parent(X,Y).
father(X,Y):-
	male(X),
	parent(X,Y).
brother(X,Y):-
	male(X),
	parent(Z,X),
	parent(Z,Y),
	X \= Y.
sister(X,Y):-
	female(X),
	parent(Z,X),
	parent(Z,Y),
	X \= Y.
daughter(X,Y):-
	female(X),
	parent(Y,X).
son(X,Y):-
	male(X),
	parent(Y,X).
grandmother(X,Y):-
	female(X),
	parent(X,Z),
	parent(Z,Y).
granfather(X,Y):-
	male(X),
	parent(X,Z),
	parent(Z,Y).
aunt(X,Y):-
	female(X),
	sibling(Z,X),
	parent(Z,Y).
uncle(X,Y):-
	male(X),
	sibling(Z,X),
	parent(Z,Y).

female(miyase).
female(ayse).
female(aydilek).
female(kezban).
female(bahriye).
female(gulden).
female(dilek).
female(deger).
female(ebru).
female(ferdane).
female(aysenur).

male(asim).
male(mustafa).
male(omer).
male(bulent).
male(halil).
male(ozal).
male(aydin).
male(arif).
male(arda).
male(muhammet).
male(hakan).

parent(miyase,aydilek).
parent(miyase,dilek).
parent(miyase,bulent).
parent(asim,aydilek).
parent(asim,dilek).
parent(asim,bulent).
parent(aydilek,ebru).
parent(aydilek,ferdane).
parent(omer,ebru).
parent(omer,ferdane).
parent(bulent,hakan).
parent(kezban,hakan).
parent(dilek,deger).
parent(dilek,arif).
parent(aydin,deger).
parent(aydin,arif).
parent(ayse,aydin).
parent(ayse,ozal).
parent(ayse,bahriye).
parent(mustafa,aydin).
parent(mustafa,ozal).
parent(mustafa,bahriye).
parent(ozal,muhammet).
parent(gulden,muhammet).
parent(bahriye,arda).
parent(bahriye,aysenur).
parent(halil,arda).
parent(halil,aysenur).

child(aydilek,miyase).
child(dilek,miyase).
child(bulent,miyase).
child(aydilek,asim).
child(dilek,asim).
child(bulent,asim).
child(ebru,aydilek).
child(ferdane,aydilek).
child(ebru,omer).
child(ferdane,omer).
child(hakan,bulent).
child(hakan,kezban).
child(deger,dilek).
child(arif,dilek).
child(deger,aydin).
child(arif,aydin).
child(aydin,ayse).
child(ozal,ayse).
child(bahriye,ayse).
child(aydin,mustafa).
child(ozal,mustafa).
child(bahriye,mustafa).
child(muhammet,ozal).
child(muhammet,gulden).
child(arda,bahriye).
child(aysenur,bahriye).
child(arda,halil).
child(aysenur,halil).

sibling(aydin,ozal).
sibling(ozal,bahriye).
sibling(bahriye,aydin).
sibling(arif,deger).
sibling(aydilek,dilek).
sibling(aydilek,bulent).
sibling(dilek,bulent).
sibling(ebru,ferdane).
sibling(aysenur,arda).

File: Hw7/code/161044096_değer_mandal.py
# Python is a imperative and functional programming language,because
# you use many commands. Python is not a functional programming language,
# but it supports some features specific to functional programming.



#       miyase+asım                              ayşe+mustafa
#             |                                      |
# -----------------------------------    -------------------------------
#   |                |            |         |      |              |
#aydilek+ömer  bülent+kezban    dilek  +  aydın  özal+gülden  bahriye+halil
# |     |           |             |          |       |         |     |
#ebru ferdane     hakan          değer      arif   muhammet  arda  ayşenur




# I used some of my family, because, I could do the homework this way.


dict_deger={"dilek":"mother",
            "aydin":"father",
            "miyase":"grandmother",
            "asim":"grandfather",
            "bulent":"uncle",
            "aydilek":"aunt",
            "arif":"brother"}

dict_miyase={"asim":"husband",
             "bulent":"son",
             "aydilek":"daughter",
             "dilek":"daughter",
             "deger":"granddaughter",
             "arif":"grandson"}

dict_asim={"miyase":"wife",
           "bulent":"son",
           "aydilek":"daughter",
           "dilek":"daughter",
           "deger":"granddaughter",
           "arif":"grandson"}

dict_arif={"dilek":"mother",
           "aydin":"father",
           "miyase":"grandmother",
           "asim":"grandfather",
           "bulent":"uncle",
           "aydilek":"aunt",
           "deger":"sister"}

dict_dilek={"miyase":"mother",
            "asim":"father",
            "aydin":"husband",
            "aydilek":"sister",
            "bulent":"brother",
            "deger":"daughter",
            "arif":"son"}



dict_aydin={"dilek":"wife",
            "deger":"daughter",
            "arif":"son"}

dict_bulent={"miyase":"mother",
             "asim":"father",
             "aydilek":"sister",
             "dilek":"sister",
             "deger":"niece",
             "arif":"nephew"}


dict_aydilek={"miyase":"mother",
             "asim":"father",
             "dilek":"sister",
             "bulent":"brother",
             "deger":"niece",
             "arif":"nephew"}

first=input("enter the first name:")
second=input("enter the second name:")

x=0
while(x<2):
    if first=="miyase":
        print(dict_miyase.get(second,"ops1"))
    if first=="asim":
        print(dict_asim.get(second,"ops2"))
    if first=="deger":
        print(dict_deger.get(second,"ops3"))
    if first=="arif":
        print(dict_arif.get(second,"ops4"))
    if first=="dilek":
        print(dict_dilek.get(second,"ops5"))
    if first=="aydin":
        print(dict_aydin.get(second,"ops6"))
    if first=="aydilek":
        print(dict_aydilek.get(second,"ops7"))
    if first=="bulent":
        print(dict_bulent.get(second,"ops8"))

    a=first
    first=second
    second=a
    del a
    x=x+1

















